code/atom.dm

```diff
@@ -238,19 +238,28 @@ TYPEINFO(/atom)
 	proc/return_air()
 		return null
 	/**
-	  * Convenience proc to see if a container is open for chemistry handling
-	  * Takes an argument of whether this openness is for the purpose of pouring something in or not (this should maybe just be a separate flag but we ran out of bits okay)
-	  * returns true if open, false if closed
+	  * Convenience proc to see if a container is open for chemistry handling.
+	  * - Takes arguments of whether this openness is for the purpose of pouring something in or out.
+	  * - `TRUE` means you're checking chems go in. `FALSE` means pouring outward. Leaving it null means checking either way.
+	  * - returns `TRUE` if you can access insides, `FALSE` if closed.
+	  * - child procs may override this for special behaviours, such as checking size.
 	  */
-	proc/is_open_container(input = FALSE)
-		return flags & OPENCONTAINER
+	proc/is_open_container(inward)
+		. = FALSE
+		if (isnull(inward) && (src.flags & ISOPEN_BOTH))
```

Comment:
Could maybe do away with this default case? It's kind of confusing and should be mostly unused

Reply:
i've made it just crash if used without overriding